controllers/projectController.js

```diff
@@ -0,0 +1,56 @@
+const knex = require("knex");
+const knexConfig = require("../knexfile");
+const db = knex(knexConfig.development);
+
+const controllersProject = {
+  addProject(req, res, next) {
+    if (!req.body.name || req.body.length <= 0) {
+      next(new Error("EMPTY BODY"));
+    }
+    const projectName = req.body;
+    db("projects")
+      .insert(projectName)
+      .then(id => res.status(201).json(id))
+      .catch(next);
+  },
+
+  addAction(req, res, next) {
+    if (req.body.length <= 0) {
+      next(new Error("EMPTY BODY"));
+    }
+    const actionName = req.body;
+    db("actions")
+      .insert(actionName)
+      .then(id => res.status(201).json(id))
+      .catch(next(new Error("ID NOT FOUND")));
```

Comment:
This should be written `.catch(()=> next(new Error("Some useful error message")));`  Same for line 52.  What's happening here is you're invoking line 25 no matter what.  The `.catch` wants a callback function declaration, not an invocation, so that it can throw the error if and only if there is one.